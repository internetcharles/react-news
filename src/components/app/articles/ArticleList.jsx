import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleElements = articles.map(article => (
    <li key={article.title}>
      <Article {...article} />
    </li>
  ));

  return (
    <ul data-testid="articles">
      {articleElements}
    </ul>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  }))
};

export default ArticleList;
