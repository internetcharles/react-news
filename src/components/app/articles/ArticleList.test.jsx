import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList', () => {
  afterEach(() => cleanup());
  it('renders ArticleList', () => {
    const { asFragment } = render(<ArticleList
      articles={[
        {
          title: 'Yo',
          author: 'It\'s',
          description: 'Lit'
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
