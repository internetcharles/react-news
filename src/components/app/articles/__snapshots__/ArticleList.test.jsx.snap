// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders ArticleList 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles"
  >
    <li>
      <figure>
        <h1>
          Yo
        </h1>
        <h3>
          It's
        </h3>
        <p>
          Lit
        </p>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
